src/WikiParsing/Articles/Article.fs

```diff
@@ -29,25 +29,29 @@ let getCondition = function
     | Starts partName -> starts partName
     | OneOf partNames -> fun x -> partNames |> Seq.contains x
 
+let isReachable (url: string) = 
+    (new HttpClient()).GetAsync(url) 
+    |> Async.AwaitTask 
+    |> Async.RunSynchronously
+    |> fun response -> response.IsSuccessStatusCode
+
 let getUrl = (+) wikiUrl
 
-let getHtml = 
-    Uri.EscapeDataString
-    >> getUrl
-    >> Article.Load
-    >> fun data -> data.Html.Descendants()
+let getArticle entry = 
+    let url = entry |> getUrl
+    if url |> isReachable
```

Comment:
I think this check could move to Article.Load assuming it's also using http client to load data. So we could have just one http call for everything

Reply:
Well, `Article.Load` is a just a Type Provider call - the type is called `Article` as well, see the file top, and `Load` is `HtmlTypeProvider` [method](https://fsharp.github.io/FSharp.Data/library/HtmlProvider.html). So I don't know how to interweave the client there...